.github/workflows/ci.yml

```diff
@@ -13,29 +13,58 @@ jobs:
     steps:
       - name: Clone This Repo
         uses: actions/checkout@v2
-      - name: Build
+      - name: Build with Sanitizers
         run: |
-          sudo apt-get install -y lcov
-          cmake -S test -B build/ \
+          sudo apt-get install -y cmake lcov
+          CFLAGS="  --coverage -O0 -Wall -Wextra"
+          CFLAGS+=" -Werror -Wno-error=pedantic"
+          CFLAGS+=" -D_FORTIFY_SOURCE=2"
+          CFLAGS+=" -Wformat"
+          CLFAGS+=" -Wformat-security"
+          CFLAGS+=" -Warray-bounds"
+          CFLAGS+=" -fsanitize=address,undefined"
+          CFLAGS+=" -fsanitize=pointer-compare -fsanitize=pointer-subtract"
+          CFLAGS+=" -fsanitize-recover=undefined"
+          CFLAGS+=" -fsanitize-address-use-after-scope"
+          CFLAGS+=" -fsanitize-undefined-trap-on-error"
+          CFLAGS_=" -fstack-protector-all"
+          cmake -S test -B build \
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
-          -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -DNDEBUG'
-          make -C build/ all
-      - name: Test
+          -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+          -DCMAKE_C_FLAGS="${CFLAGS}"
+          make -C build all
+      - name: Run Tests with Sanitizers
         run: |
-          cd build/
+          cd build
+          make coverage
```

Comment:
(Minot) What's the purpose of building `coverage` target instead of just the unit test binaries with the `all` target when the next step in this job runs test for coverage?

Reply:
You don't need the `coverage` target to build and run the unit tests necessarily. The `ctest` command takes care of running the unit tests